apps/generic_debugger.talon

```diff
@@ -1,73 +1,59 @@
 tag: debugger
 -
-
 # Code execution
-
-## instruction level
+# instruction level
 step into: user.debugger_step_into()
 step over: user.debugger_step_over()
-
-## line level
+# line level
```

Comment:
I don't like how two ##s are being replaced by one # here. The use of ## seems to have been deliberate. Comments are for expressing human intent, so I don't think auto formatters should change them.

Reply:
> I'm against changing comments by adding a leading space, though. Comments are for expressing human intent so I think that autoformatters should leave them alone.

Code formatting expresses human intent. The purpose of a formatter is to enforce some measure of consistency. However, I've made you a "--no-format-comments" flag, so we can at least not hardcode this behaviour.

Reply:
> The PR looks much better now that it tries to put things on a single line where possible. It is removing the nice horizontal alignment some .talon files have, though. I remember there was some discussion on the corresponding issue of the right way to do this to avoid large diffs - how does talonfmt handle alignment?

There's automatic alignment of one-line commands with `--align-short-commands` and fixed column alignment of one-line commands with, e.g., `--align-short-commands-at=32`. I've not activated the former, due to the talk about big diffs, and not activated the latter because I think it's ugly.

The same commands exist for matches in the context header (though I don't think `--align-context-matches` actually works at the moment, since I never bothered too much with it, but it should be a single function call away).

Reply:
Created #949 and #950 to showcase these variants, but I'd prefer to not use either.